dIM/Views/Home/SettingsView/SettingsView.swift

```diff
@@ -50,101 +50,257 @@ struct SettingsView: View {
+                                self.storedUsername = usernameTemp
+                                removeIDfromUsername()
+                                if usernameTemp == editableUsername + "#" + userID {
+                                    self.hasChangedUsername = false
+                                }
+                            }
+                            .keyboardType(.namePhonePad)
+                            .padding()
+                            .cornerRadius(10.0)
+                            Text("#" + userID)
+                                .foregroundColor(.gray)
                             
-                            if checkValidUsername(username: usernameTemp) {
-                                let usernameDigits = usernameTemp + "#" + String(Int.random(in: 100000...999999))
-                                defaults.set(usernameDigits, forKey: "Username")
-                            } else {
-                                usernameTemp = ""
+                            Spacer()
+                            if hasChangedUsername {
+                                Button(action: {
+                                    self.wantsToRevertUsernameChange = true
+                                }, label: {
+                                    Image(systemName: "arrow.uturn.left")
+                                })
+                                .disabled(checkValidUsername(username: editableUsername) == false)
                             }
+                            
                         }
-                    )
-                    .keyboardType(.namePhonePad)
-                    .padding()
-                    .background(
-                        colorScheme == .dark ? Color("setup-grayDARK") : Color("setup-grayLIGHT")
-                    )
-                    .cornerRadius(10.0)
+                        .foregroundColor(.accentColor)
+                    }
                     
-                    /*
-                     Text below textfield.
-                     */
-                    if usernameTemp.count < 4 {
-                        Text("Minimum 4 characters.")
-                            .font(.footnote)
-                            .foregroundColor(.accentColor)
-                    } else if usernameTemp.count > 16 {
-                        Text("Maximum 16 characters.")
-                            .font(.footnote)
-                            .foregroundColor(.accentColor)
-                    } else if usernameTemp.contains(" ") {
-                        Text("No spaces in username.")
-                            .font(.footnote)
+                    
+                } header: {
+                    Text("My Username")
+                } footer: {
+                    Text("If you change your username, you and your contacts will have to add each other again.")
+                }
+                Section {
+                    Toggle(isOn: .constant(true), label: {
```

Comment:
`Toggle` is currently always on 🤔

Reply:
Whoops, left it there for testing. Will remove it once I get home.